### ISSUE TITLE
Duplicate GPS track log points being recorded

### ISSUE THREAD

**Issue by AndrewM-** [geopaparazzi/geopaparazzi#581]

**Comment by AndrewM-** [comment]
Hi @moovida, in the above screen clip, I am actually in a gpap file and looking at the track log.  I thought that I might have got the same points by going up and down the same trail but the timestamps would be different. I will try to contribute a query that can be run directly on the sqlite file to detect duplicate points.  I imagine that you can do this probably easier than I can.  As I have put a distinct row clause in my algorithm, this does issue does not bother me anymore. I was just reporting it.

To answer your question, I am not merging files and I should only have one instance of geopaparazzi running so should not be recording the point twice.  Also only some of the points in the table are duplicated.  I will look into it further.  I wonder if the device is giving geopaparazzi bad data?

**Comment by moovida** [comment]
Hi @AndrewM- , thanks for this report. If what you say is correct and the above is part of the same gpap file, then we are looking at a bad bad bug. 
If the points are of two different logs and only half a second away (while 12 logs have been happening between log 6 and 18), then something is very wrong. Also the exact same coordinate looks disturbing.

If you are willing to send me the database, I would be interested to look at it.

@brentfraser , a pair of experienced eyes would sure help. Do you have any idea about this?

**Comment by AndrewM-** [comment]
I will send you a gpap file that has the issue now.  I am using dbeaver and can check for issues with the following query which will return the duplicated points.

> Select lat, lon, count(lon) from gpslogsdata group by lat having count(lon) > 1

**Comment by AndrewM-** [comment]
I have looked at 2 gpap projects using QGIS as viewer.   In both of these files, a track segment is duplicated and the duplicated track is several track segments after the original.  Both files were multi-day files and I wonder if I have not been exiting the application properly and causing this issue.  It could be a sqlite issue.  In both cases, the entire track segment is copied and there is no data missing.  In using QGIS as a viewer, I just change the file extension to sqlite, then bring the gpslogsdata in as a non-spatial table then run the create points process to make the points on the map.

**Comment by moovida** [comment]
@AndrewM- , sorry for the delay, have been weeks of travel. I have looked at your database and can't really understand what is going on. I have no database that suppers from duplicated points and am not able to identify where things might go wrong. Do you have any idea how this could be debugged?

**Comment by AndrewM-** [comment]
It is always hard to track these ones down.  Perhaps you can add a diagnostics menu to the application that checks the file for duplicates.  I use accounting software that can check the integrity of the database.  The integrity checker could also be external.  Basically I have one now with ms-access or dbeaver, with queries that detect issues.  I think we have to get more examples of the issue occurring.  I have upgraded to the current version of Geopaparazzi and will also be promoting it on Wed. this week to our local GIS group.  The more users, the faster these rare issues will be found.  I think the main impact of the issue is that GPX files are hard to import into ArcMap if there are duplicate points.

**Comment by moovida** [comment]
Hi @AndrewM- , this is actually a brilliant idea. We could create a wiki page with a table containing:

* rule label/title
* rule query 
* expected result

I could then create something in geopaparazzi that can be run at any time and would execute these check queries and a report tels the user about failing checks. What do you think?

**Comment by AndrewM-** [comment]
Yes, that would help to isolate tricky bugs more quickly. I look forward to it.

**Comment by moovida** [comment]
@AndrewM- what would be best to start? A wiki page with the table? Or should we use an issue? I think the wiki is better because one will also see what checks are implemented already.

If you start the wiki page with a set of queries you would like to have implemented, I will make sure to get this working in geopap asap.

**Comment by AndrewM-** [comment]
@moovida Just started a new job at the other end of my country so I have been a little quiet.  Have been recording some more trials to see if the issue recurs.  I could do the wiki, but it would take a few weeks.
